client/src/dataset/validate.ts

```diff
@@ -17,7 +17,22 @@ export function convertErrors(errors: ErrorObject[]): FormikErrors<FormikValues>
     return res;
 }
 
-export async function validateOpen(type: string, data: object) {
+export function throwErrors(validateErrors: ErrorObject[] | undefined | null, customValidateErrors?: FormikErrors<FormikValues>) {
+    if (validateErrors && customValidateErrors) {
+        const converted = { ...convertErrors(validateErrors), customValidateErrors };
+        throw converted;
+    } else if (customValidateErrors) {
+        const converted = customValidateErrors;
+        throw converted;
+    }  else if (validateErrors) {
+        const converted = convertErrors(validateErrors);
+        throw converted;
+    } else {
+        throw new Error("unspecified error while validating fields");
+    }
+}
```

Comment:
i think this can be shortened by having an empty default value for the custom validation errors:

```suggestion
export function throwErrors(validateErrors : ErrorObject[] | null = [], customValidateErrors: FormikErrors<FormikValues> = {}) {
    if (validateErrors) {
        const converted = { ...convertErrors(validateErrors), customValidateErrors };
        throw converted;
    } else {
        throw new Error("unspecified error while validating fields");
    }
}
```

What do you think?

Reply:
Thanks! I also made a few changes on top of your suggested way.